viewer/packages/camera-manager/src/StationaryCameraManager.ts

```diff
@@ -181,7 +206,40 @@ export class StationaryCameraManager implements CameraManager {
     this._camera.quaternion.setFromEuler(euler);
 
     this._cameraChangedListeners.forEach(cb => cb(this._camera.position, this.getTarget()));
-  };
+  }
+
+  private pinchCamera(moveEvent: PointerEvent) {
+    if (this._downEventCache.length < 2) {
+      return;
+    }
+
+    const getEuclideanDistance = (eventOne: PointerEvent, eventTwo: PointerEvent): number => {
+      const dx = eventOne.clientX - eventTwo.clientX,
+        dy = eventOne.clientY - eventTwo.clientY;
+      return Math.sqrt(dx * dx + dy * dy);
+    };
+
+    const preMoveDistance = getEuclideanDistance(this._downEventCache[0], this._downEventCache[1]);
+    const downEventIndex = this._downEventCache.findIndex(event => event.pointerId === moveEvent.pointerId);
+    const preMoveEvent = this._downEventCache[downEventIndex];
+    this._downEventCache[downEventIndex] = moveEvent;
+    const postMoveDistance = getEuclideanDistance(this._downEventCache[0], this._downEventCache[1]);
+    const distanceDelta = postMoveDistance - preMoveDistance;
```

Comment:
Concider moving this out (or parts of it) into a separate function. A little hard to follow what is going on.

Reply:
Added calculateDownEventDistance(). 👑